source/fuzz/transformation_wrap_vector_synonym.cpp

```diff
@@ -0,0 +1,163 @@
+  if (instruction == nullptr) {
+    return false;
+  }
+
+  std::unordered_set<SpvOp_> valid_opcodes{SpvOpIAdd, SpvOpISub, SpvOpIMul,
+                                           SpvOpSDiv, SpvOpUDiv, SpvOpFAdd,
+                                           SpvOpFSub, SpvOpFMul, SpvOpFDiv};
+
+  if (!valid_opcodes.count(instruction->opcode())) {
+    return false;
+  }
+
+  // |vec_type_id| must refer to a valid type instruction.
+  if (vector_type == nullptr) {
+    return false;
+  }
+
+  // |result_id1| and |result_id2| must exists.
+  if (vec1 == nullptr || vec2 == nullptr) {
+    return false;
+  }
+
+  // |vector_type_id| must correspond to a valid vector type.
+  if (vector_type->AsVector() == nullptr) {
+    return false;
+  }
+
+  // |vec_id| must be fresh.
+  if (!fuzzerutil::IsFreshId(ir_context, message_.vec_id())) {
+    return false;
+  }
+
+  std::vector<uint32_t> vec_ids{message_.result_id1(), message_.result_id2(),
+                                message_.vec_id()};
+
+  // |vec_id|, |result_id1| and |result_id2| should be disparate ids.
+  if (fuzzerutil::HasDuplicates(vec_ids)) {
+    return false;
+  }
+
+  // |scalar_position| needs to be a non-negative integer less than the vector
+  // length.
+  auto vec_len = ir_context->get_def_use_mgr()
+                     ->GetDef(message_.vec_type_id())
+                     ->GetSingleWordOperand(1);
+  if (message_.scalar_position() >= vec_len) {
+    return false;
+  }
+
+  // The 2 vectors must have the same type as the result vector type.
+  auto vec1_type_id = vec1->GetSingleWordOperand(0);
+  auto vec2_type_id = vec2->GetSingleWordOperand(0);
+
+  if (vec1_type_id != message_.vec_type_id() ||
+      vec2_type_id != message_.vec_type_id()) {
+    return false;
+  }
+
+  return true;
+}
+
+void TransformationWrapVectorSynonym::Apply(
+    opt::IRContext* ir_context,
+    TransformationContext* transformation_context) const {
+  auto instruction =
+      ir_context->get_def_use_mgr()->GetDef(message_.instruction_id());
+  // Create an instruction descriptor for the original instruction.
+  auto inst_descriptor = MakeInstructionDescriptor(message_.instruction_id(),
+                                                   instruction->opcode(), 0);
+  auto insert_before_inst = FindInstruction(inst_descriptor, ir_context);
+  auto destination_block = ir_context->get_instr_block(insert_before_inst);
+  auto insert_before = fuzzerutil::GetIteratorForInstruction(
+      destination_block, insert_before_inst);
+
+  //  Populate input operand list with two vectors for vector operation.
+  opt::Instruction::OperandList in_operands;
+  in_operands.push_back({SPV_OPERAND_TYPE_ID, {message_.result_id1()}});
+  in_operands.push_back({SPV_OPERAND_TYPE_ID, {message_.result_id2()}});
+
+  // Make a new arithmetic instruction: %vec_id = OpXX %type_id %result_id1
+  // %result_id2.
+  auto new_instruction = MakeUnique<opt::Instruction>(
+      ir_context, instruction->opcode(), message_.vec_type_id(),
```

Comment:
Once you remove vec_type_id from the message, you'll instead need to use the result type of one of the vector operands here.

Reply:
Done.